src/litegraph.js

```diff
@@ -6052,9 +6053,7 @@ LGraphNode.prototype.executeAction = function(action)
 							this.graph.beforeChange();
                             this.node_dragged = node;
                         }
-                        if (!this.selected_nodes[node.id]) {
-                            this.processNodeSelected(node, e);
```

Comment:
Why removing this check?

Reply:
he is doing it from inside the node using this.is_selected, which I dont like so much but I guess its fine

Reply:
mhh.. what about the events connected?

Reply:
yeah, you are right, it could be a problem, I will give it a look

Reply:
I had a little try and it seems that unselecting (cntrl click on a selected node) is not actually working/implemented in the demo sites.
Was it never active before?
So it would be good to implement it!
Just let's do it properly and create the same structure as for selecting.

Reply:
as you mentioned, I removed the check to enable blocks to get unselected. In some cases, if the user drags or double-clicks a block, we'll have an **unselected** block dragging or opening a panel.

your note convinced me to review the code in order to solve the problem. I've fixed them and will send a PR, but I'm still unsure how the `onNodeSelected` event may cause a problem, so please review my next PR for any potential issues.

Reply:
Without that `processNodeSelected` is called, that itself calls `selectNode` and than an eventual existing `onNodeSelected`. So if we use processNodeSelected to toggle the selection that would not be correct.
There is an existing `deselectNode` function with his events. That function is for a single node but it's already there and if needed the multi node could be implemented as for selection.
From my view to do things correctly there should be a new method to toggle node selections that would than call `selectNode` or `deselectNode` coherently.

Reply:
I agree with @atlasan